packages/api/src/schema/api-schema.yaml

```diff
@@ -2138,11 +2544,198 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/error"
+  /asset/request-upload:
+    post:
+      tags:
+        - Asset
+      summary: Upload an asset
+      description: >
+        To upload an asset, your first need to request for a direct upload URL
+        and only then actually upload the contents of the asset.
+
+
+        This also allows your users to upload files since the upload URL is
+        pre-signed with your credentials and is accessible from the browser. The
+        initial API call to get the upload URL must be done from a backend
+        though, from which an API token is available.
+
+
+        Once you created a upload link, you have 2 options, resumable or direct
+        upload. For a more reliable experience, you should use resumable uploads
+        which will work better for users with unreliable or slow network
+        connections. If you want a simpler implementation though, you should
+        just use a direct upload.
+
+
+        ## Direct Upload
+
+        For a direct upload, make a PUT request to the URL received in the url
+        field of the response above, with the raw video file as the request
+        body. response above:
+
+
+        ## Resumable Upload
+
+        Livepeer supports resumable uploads via Tus(opens in a new tab). This
+        section provides a simple example of how to use tus-js-client to upload
+        a video file.
+
+
+        From the previous section, we generated a URL to upload a video file to
+        Livepeer on POST /api/asset/request-upload. You should use the
+        tusEndpoint field of the response to upload the video file and track the
+        progress:
+
+
+        ```javascript 
```

Comment:
This is good for now, but we definitely need to move this to a guide/tutorial. Now that we have these auto-generated API reference we should make it more objective and technical.